src/sparseml/pytorch/image_classification/train.py

```diff
@@ -668,27 +618,70 @@ def main():
     training_args, _ = _parser.parse_args_into_dataclasses()
 
     save_dir, loggers = helpers.get_save_dir_and_loggers(
-        training_args, task=CURRENT_TASK
+        args=training_args, task=CURRENT_TASK
     )
 
-    input_shape = ModelRegistry.input_shape(training_args.arch_key)
-    image_size = input_shape[1]  # assume shape [C, S, S] where S is the image size
-
     (
         train_dataset,
         train_loader,
         val_dataset,
         val_loader,
-    ) = helpers.get_train_and_validation_loaders(
-        training_args, image_size, task=CURRENT_TASK
-    )
+    ) = helpers.get_train_and_validation_loaders(args=training_args, task=CURRENT_TASK)
 
-    num_classes = helpers.infer_num_classes(training_args, train_dataset, val_dataset)
+    num_classes = helpers.infer_num_classes(
+        args=training_args,
+        train_dataset=train_dataset,
+        val_dataset=val_dataset,
+    )
 
-    # # model creation
-    model = helpers.create_model(training_args, num_classes)
     train(
-        training_args, model, train_loader, val_loader, input_shape, save_dir, loggers
+        train_args=training_args,
+        num_classes=num_classes,
+        train_loader=train_loader,
+        val_loader=val_loader,
+        save_dir=save_dir,
+        loggers=loggers,
+    )
+
+
+def _initialize_trainer(
+    train_args: TrainingArguments,
+    train_loader: DataLoader,
+    val_loader: DataLoader,
+    loggers: List[Any],
+    num_classes: int,
+) -> ImageClassificationTrainer:
+    """
+    Initialize and return the trainer
+    """
+    extras = {"top1acc": TopKAccuracy(1), "top5acc": TopKAccuracy(5)}
+
+    def loss_fn():
+        return CrossEntropyLossWrapper(extras)
+
+    model, key = helpers.create_model(
+        args=train_args,
+        num_classes=num_classes,
+    )
+    LOGGER.info(f"created model with key {key}: {model}")
+
+    return ImageClassificationTrainer(
+        model=model,
+        key=key,
+        recipe_path=train_args.recipe_path,
+        local_rank=train_args.local_rank,
```

Comment:
args look better, why do we still need rank and local rank if we have `device` and `is_main_process`?